packages/botonic-plugin-contentful/src/contentful/contents/handoff.ts

```diff
@@ -0,0 +1,118 @@
+import * as contentful from 'contentful'
+
+import * as cms from '../../cms'
+import {
+  ContentType,
+  HandoffAgentEmail,
+  HandoffAgentId,
+  HandoffDestination,
+  HandoffQueue,
+  OnFinish,
+} from '../../cms'
+import { DeliveryApi } from '../delivery-api'
+import {
+  addCustomFields,
+  CommonEntryFields,
+  ContentfulEntryUtils,
+} from '../delivery-utils'
+import { CallbackTarget, getTargetCallback } from './callback-delivery'
+import { DeliveryWithFollowUp } from './follow-up'
+import { QueueDelivery, QueueFields } from './queue'
+
+export class HandoffDelivery extends DeliveryWithFollowUp {
+  static REFERENCES_INCLUDE = QueueDelivery.REFERENCES_INCLUDE + 1
+  constructor(
+    delivery: DeliveryApi,
+    private readonly queue: QueueDelivery,
+    resumeErrors: boolean
+  ) {
+    super(ContentType.HANDOFF, delivery, resumeErrors)
+  }
+
+  async handoff(id: string, context: cms.Context): Promise<cms.Handoff> {
+    const entry: contentful.Entry<HandoffFields> = await this.getEntry(
+      id,
+      context,
+      { include: HandoffDelivery.REFERENCES_INCLUDE }
+    )
+    return this.fromEntry(entry, context)
+  }
+
+  private onFinish(
+    entry: contentful.Entry<HandoffFields>,
+    context: cms.Context
+  ): OnFinish | undefined {
+    if (entry.fields.onFinish) {
+      return getTargetCallback(entry.fields.onFinish, context)
+    }
+    return undefined
+  }
+
+  private destination(
+    entry: contentful.Entry<HandoffFields>
+  ): HandoffDestination | undefined {
+    if (!entry.fields.destination) return undefined
+    const AGENT_EMAIL_TYPE = 'agentEmail'
+    const AGENT_ID_TYPE = 'agentId'
```

Comment:
I've seen the string literals 'agentEmail' and 'agentId' being used somewhere else in this code. Will it make sense to use them always as constants?